File: ddayapi/models/remindermodel.js
// models/remindermodel.js
const mongoose = require('mongoose');

const reminderSchema = new mongoose.Schema({
  firm_name: { type: String, required: true },
  firm_address: { type: String, required: true },
  gst: { type: String, required: true },
  certification_body: { type: String, required: true },
  contact: { type: Number, required: true },
  basic_amount: { type: Number, required: true },
  reference: { type: String, required: true },
  issued_date: { 
    type: Date, 
    required: true,
    get: function (value) {
      return value ? new Date(value).toLocaleDateString('en-GB') : ''; // 'en-GB' is for dd-MM-yyyy format
    }
  },
  expiry_date: { 
    type: Date, 
    required: true,
    get: function (value) {
      return value ? new Date(value).toLocaleDateString('en-GB') : '';
    }
  },
  certificate: { type: String, required: true },
});

// To make sure the getter works when fetching data
reminderSchema.set('toJSON', { getters: true });

const Reminder = mongoose.model('Reminder', reminderSchema);

module.exports = Reminder;

File: lib/model/remindermodel.dart
class ReminderModel {
  final bool? success;
  final List<Reminder>? reminders;

  ReminderModel({this.success, this.reminders});

  factory ReminderModel.fromJson(Map<String, dynamic> json) {
    return ReminderModel(
      success: json['success'],
      reminders: (json['reminders'] as List<dynamic>?)
          ?.map((e) => Reminder.fromJson(e))
          .toList(),
    );
  }
}

class Reminder {
  final String? id; // Added to handle the `_id` field
  final String? firmName;
  final String? firmAddress;
  final String? gst;
  final String? certificationBody;
  final String? contact;
  final double? basicAmount;
  final String? reference;
  final String? issuedDate;
  final String? expiryDate;
  final String? certificate;

  Reminder({
    this.id, // Added to constructor
    this.firmName,
    this.firmAddress,
    this.gst,
    this.certificationBody,
    this.contact,
    this.basicAmount,
    this.reference,
    this.issuedDate,
    this.expiryDate,
    this.certificate,
  });

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
      id: json['_id'], // Correctly mapping `_id` to `id`
      firmName: json['firm_name'],
      firmAddress: json['firm_address'],
      gst: json['gst'],
      certificationBody: json['certification_body'],
      contact: json['contact'].toString(), // Contact converted to string
      basicAmount: double.tryParse(json['basic_amount'].toString()),
      reference: json['reference'],
      issuedDate: json['issued_date'],
      expiryDate: json['expiry_date'],
      certificate: json['certificate'],
    );
  }
}
